### PR TITLE
Add basic WDL development support

### LINKED ISSUE #3384: Update WDL lexer
The WDL version needs to be updated in order to support the newer standard library functions as discussed in https://github.com/DataBiosphere/toil/pull/3364#discussion_r539558414.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-762)
┆Issue Number: TOIL-762

### PR DESCRIPTION
**Pull request by w-gao**
This can probably close #3384, but we still need to implement the following to fully support the development version of WDL: 

- [Import statements](https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#import-statements)
- [Directory type](https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#types)
- [Structs](https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#struct-definition)
- [`after` keyword in call statements](https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#call-statement)

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Add basic WDL development support.

### CODE REVIEW

**Review by adamnovak — APPROVED** [review]
This looks good to me. We do still need the implementations for some new features, as you note. Do we have issues for those?

**Comment by w-gao** [comment]
@adamnovak Thanks for the review! We do not have issues for them yet.

File: src/toil/wdl/versions/dev.py [hunk 1]
```diff
@@ -33,4 +39,68 @@ def analyze(self):
         intermediate data structures: `self.workflows_dictionary` and
         `self.tasks_dictionary`.
         """
-        raise NotImplementedError
+        lexer = WdlLexer(FileStream(self.wdl_file))
+        parser = WdlParser(input=CommonTokenStream(lexer))
+        tree = parser.document()
+        self.visit_document(tree)
+
+    def visit_document(self, ctx):
```
Comment (DailyDreaming) [inline src/toil/wdl/versions/dev.py:47]: Could type hinting be added here and to the rest of the added functions?

**Review by DailyDreaming — APPROVED** [review]
LGTM, though see comment.

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thanks for adding the type hinting.

**Pull request merged**